core_impl/src/main/java/io/opencensus/stats/MutableViewData.java

```diff
@@ -15,179 +15,232 @@
-//    }
-//
-//    @Override
-//    void record(StatsContextImpl tags, long value) {
-//      throw new UnsupportedOperationException("Not implemented.");
-//    }
-//
-//    @Override
-//    final ViewData toViewData(Clock clock) {
-//      throw new UnsupportedOperationException("Not implemented.");
-//    }
-//  }
+  @VisibleForTesting
+  static List<TagValue> getTagValues(Map<TagKey, TagValue> tags, List<TagKey> columns) {
+    List<TagValue> tagValues = new ArrayList<TagValue>(columns.size());
+    // Record all the measures in a "Greedy" way.
+    // Every view aggregates every measure. This is similar to doing a GROUPBY view’s keys.
+    for (int i = 0; i < columns.size(); ++i) {
+      TagKey tagKey = columns.get(i);
+      if (!tags.containsKey(tagKey)) {
+        // replace not found key values by “unknown/not set”.
+        tagValues.add(UNKNOWN_TAG_VALUE);
+      } else {
+        tagValues.add(tags.get(tagKey));
+      }
+    }
+    return tagValues;
+  }
+
+  /**
+   * Create an empty {@link MutableAggregation} based on the given {@link Aggregation}.
+   *
+   * @param aggregation {@code Aggregation}.
+   * @return an empty {@code MutableAggregation}.
+   */
+  @VisibleForTesting
+  static MutableAggregation createMutableAggregation(Aggregation aggregation) {
+    return aggregation.match(
+        new Function<Sum, MutableAggregation>() {
+          @Override
+          public MutableAggregation apply(Sum arg) {
+            return MutableSum.create();
+          }
+        },
+        new Function<Count, MutableAggregation>() {
+          @Override
+          public MutableAggregation apply(Count arg) {
+            return MutableCount.create();
+          }
+        },
+        new Function<Histogram, MutableAggregation>() {
+          @Override
+          public MutableAggregation apply(Histogram arg) {
+            return MutableHistogram.create(arg.getBucketBoundaries());
+          }
+        },
+        new Function<Range, MutableAggregation>() {
+          @Override
+          public MutableAggregation apply(Range arg) {
+            return MutableRange.create();
+          }
+        },
+        new Function<Mean, MutableAggregation>() {
+          @Override
+          public MutableAggregation apply(Mean arg) {
+            return MutableMean.create();
+          }
+        },
+        new Function<StdDev, MutableAggregation>() {
+          @Override
+          public MutableAggregation apply(StdDev arg) {
+            return MutableStdDev.create();
+          }
+        },
+        Functions.<MutableAggregation>throwIllegalArgumentException());
+  }
+
+  /**
+   * Create an {@link AggregationData} snapshot based on the given {@link MutableAggregation}.
+   *
+   * @param aggregation {@code MutableAggregation}
+   * @return an {@code AggregationData} which is the snapshot of current summary statistics.
+   */
+  @VisibleForTesting
+  static AggregationData createAggregationData(MutableAggregation aggregation) {
```

Comment:
Either adding a `toAggregationData()` method or having a `createAggregationData()` method that does the type matching is fine. I slightly prefer to keep using the match-and-create method since it's more consistent with creating `MutableAggregation` from `Aggregation`.